src/ccontrol/UserQuerySelect.h

```diff
@@ -158,7 +158,7 @@ class UserQuerySelect : public UserQuery {
     void _expandSelectStarInMergeStatment(std::shared_ptr<query::SelectStmt> const& mergeStmt);
 
     void _discardMerger();
-    void _qMetaUpdateStatus(qmeta::QInfo::QStatus qStatus, size_t rows, size_t bytes);
+    void _qMetaUpdateStatus(qmeta::QInfo::QStatus qStatus, size_t rows, size_t bytes, size_t finalRows);
```

Comment:
Since you don't have any documentation for the parameters of the method it would be helpful to give more descriptive names to the parameters:
```
    void _qMetaUpdateStatus(qmeta::QInfo::QStatus qStatus,
                            size_t collectedRows = 0,
                            size_t collectdBytes = 0,
                            size_t finalRows = 0);
```
This will make it easier for viewers to trace all uses (initializations, etc.) of these parameters in the code.
And if you make the last 3 parameters optional then you may not need to explicitly pass zeroes for those in the 3 calls mentioned earlier in this review.